internal/storage/image.go

```diff
@@ -151,6 +152,14 @@ type ImageServer interface {
 	// CandidatesForPotentiallyShortImageName resolves an image name into a set of fully-qualified image names (domain/repo/image:tag|@digest).
 	// It will only return an empty slice if err != nil.
 	CandidatesForPotentiallyShortImageName(systemContext *types.SystemContext, imageName string) ([]RegistryImageReference, error)
+	// ImageRemovalLock returns the removal lock for the specified image storage reference.
+	ImageRemovalLock(id StorageImageID) *sync.RWMutex
+}
+
+// ImageRemovalLock returns the removal lock for the specified image storage reference.
+func (svc *imageService) ImageRemovalLock(id StorageImageID) *sync.RWMutex {
+	value, _ := svc.imageRemovalLock.LoadOrStore(id, &sync.RWMutex{})
```

Comment:
@haircommander, it's not an error.

It's a boolean to indicate whether the value was loaded or stored. In case you need to know which operations succeeded.

Reply:
RTFM @haircommander :)